sovtokenfees/sovtokenfees/fees_authorizer.py

```diff
@@ -12,31 +15,79 @@
 
 from state.pruning_state import PruningState
 
+from plenum.common.constants import TXN_TYPE
+
 logger = getlogger()
 
 
 class FeesAuthorizer(AbstractAuthorizer):
-    def __init__(self, fees_req_handler: StaticFeesReqHandler, config_state: PruningState):
+    def __init__(self,
+                 fees_req_handler: StaticFeesReqHandler,
+                 config_state: PruningState,
+                 utxo_cache: UTXOCache):
         super().__init__()
         self.fees_req_handler = fees_req_handler
         self.config_state = config_state
+        self.utxo_cache = utxo_cache
         self.state_serializer = StaticFeesReqHandler.state_serializer
 
-    def _get_fees_alias_from_constraint(self, auth_constaint: AuthConstraint):
-        if auth_constaint.metadata:
-            if FEES_FIELD_NAME in auth_constaint.metadata:
-                return auth_constaint.metadata[FEES_FIELD_NAME]
-
-    def _can_pay_fees(self, request, required_fees):
+    def can_pay_fees(self, request, required_fees):
         try:
-            self.fees_req_handler.can_pay_fees(request, required_fees)
+            self._can_pay_fees(request, required_fees)
         except Exception as e:
             return False, str(e)
+
         return True, ''
 
+    def _can_pay_fees(self, request, required_fees):
+
+        if request.operation[TXN_TYPE] == XFER_PUBLIC:
+            # Fees in XFER_PUBLIC is part of operation[INPUTS]
+            inputs = request.operation[INPUTS]
+            outputs = request.operation[OUTPUTS]
+            self._validate_fees_can_pay(request, inputs, outputs, required_fees)
+            self.fees_req_handler.deducted_fees_xfer[request.key] = required_fees
```

Comment:
It's better if we keep `deducted_fees_xfer` in `StaticFeesReqHandler`. Keep it there as a hook for `NodeHooks.PRE_DYNAMIC_VALIDATION` which will be replaced by Pluggable request handlers later?